WordPress/Classes/Models/BlockEditorSettingElement+CoreDataProperties.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import CoreData
+
+enum BlockEditorSettingElementTypes: String {
+    case color
+    case gradient
+
+    var valueKey: String {
+        self.rawValue
+    }
+}
+
+extension BlockEditorSettingElement {
+
+    @nonobjc public class func fetchRequest() -> NSFetchRequest<BlockEditorSettingElement> {
+        return NSFetchRequest<BlockEditorSettingElement>(entityName: "BlockEditorSettingElement")
+    }
+
+    /// Stores the associated type that this object represents.
+    /// Available types are defined in `BlockEditorSettingElementTypes`
+    ///
+    @NSManaged public var type: String
+
+    /// Stores the value for the associated type. The associated field in the API response might differ based on the type.
+    ///
+    @NSManaged public var value: String
+
+    /// Stores a unique key associated to the `value`.
+    ///
+    @NSManaged public var slug: String
```

Comment:
I appreciate that this is being implemented in a somewhat general way, with thought of future settings. I recently learned that there will be a need for an experimental feature flag setting for the Gallery block refactor work as well: https://github.com/WordPress/gutenberg/pull/25940/files#diff-e6bd94e1847aea4b70a6cc1ac9b2471dd3d882f6cbf1ab51438e23b02ca5194fR114-R128, but I'm not sure it fits this model, since there is no slug. Then again, there is no REST API yet for that, so perhaps `id` could be passed as slug :thinking: .

In any case, I'm mentioning here because there seem to a be a few upcoming cases with similar requirements (patterns will also need something similar, for example).

Reply:
> So, the element is meant to be general, but only a part of the `BlockEditorSettings`, which could easily store the additional experimental flag as well. Does that sound right?

Yup this is meant to be a more generic container for the arrays that come down from the call `/wp/v2/themes?status=active` and the one being added to `/wp-block-editor/v1/settings` in https://github.com/wordpress-mobile/gutenberg-mobile/issues/3163 the parent object is the `BlockEditorSettings` 

The main uses I see for `BlockEditorSettingElement` are colors, gradients, and font sizes. If we got to a use case that doesn't have a need for one of the required values then we can provide an empty string. This would all be about how the consumer decides to use the stored data. 

>I recently learned that there will be a need for an experimental feature flag setting for the Gallery block refactor work as well ... Then again, there is no REST API yet for that, so perhaps `id` could be passed as `slug` 🤔 .

Let's dig into this use case more once we have an idea about how it will be delivered to the app. If it's included in the call `/wp-block-editor/v1/settings`  my recommendation would be to add it to `BlockEditorSettings` but let's see.